### PR TITLE
Hover plugin enhancements

### PR DESCRIPTION
**Pull request by dnlkoch**
Here you'll find some enhancements regarding the hover plugin class:
* Prettify indenting (no functional enhancement...)
* Introduce `dynamicHoverColor` config. If set to true, the hovered feature color will be determined by the currently hovered pixel color value.
* Add hoverTemplate strings, that can be used to create more complex feature info popup contents, e.g. to have a combination of different feature properiers. See new method `getHoverTextFromTemplate()`.

### CODE REVIEW

File: src/plugin/Hover.js [hunk 1]
```diff
@@ -162,394 +179,517 @@ Ext.define('BasiGX.plugin.Hover', {
...
+     *
+     */
+    transparify: function(baseColor, alpha) {
+        var rgbaTemplate = 'rgba({0}, {1}, {2}, {3})';
+        var fallbackBaseColor = [255, 0, 0];
+        var red;
+        var green;
+        var blue;
+
+        if (baseColor instanceof Uint8ClampedArray && baseColor.length === 4 &&
```
Comment (marcjansen) [inline src/plugin/Hover.js]: IE 9 or 10 won't have `Uint8ClampedArray`, can we guard against this?

File: src/plugin/Hover.js [hunk 2]
```diff
@@ -162,394 +179,517 @@ Ext.define('BasiGX.plugin.Hover', {
...
-       me.getHoverVectorLayerSource().clear();
-       map.getOverlays().forEach(function(o) {
-           if (o.get(overlayIdentifierKey) === overlayIdentifierVal) {
-               map.removeOverlay(o);
-           }
-       });
-   },
+        if (!hoverVectorLayer) {
+            hoverVectorLayer = new ol.layer.Vector({
+                name: 'hoverVectorLayer',
```
Comment (marcjansen) [inline src/plugin/Hover.js]: Does this need to have the `BasiGX.util.Layer.KEY_DISPLAY_IN_LAYERSWITCHER` property set to false?

File: src/plugin/Hover.js [hunk 3]
```diff
@@ -162,394 +179,517 @@ Ext.define('BasiGX.plugin.Hover', {
...
+        } else if (count) {
+            radius = count * 2;
+            fontSize = count * 1.3;
+        }
+
+        return [
+            new ol.style.Style({
+                fill: new ol.style.Fill({
+                    color: me.getDynamicHoverColor() ?
+                            me.transparify(baseColor, 0.3) : hoverColor
```
Comment (marcjansen) [inline src/plugin/Hover.js]: Can we put `me.getDynamicHoverColor()` and `me.transparify(baseColor, 0.3)` into variables and reuse these?

File: src/plugin/Hover.js (hunk 2 above)
Comment (dnlkoch) [inline src/plugin/Hover.js]: See line [224](https://github.com/terrestris/BasiGX/pull/120/files/4a3dc4a15e3d58af532a59353623cdc2abdf802b#diff-175d1462889fb2cd4cfcd4584e3b1017R224).

File: src/plugin/Hover.js (hunk 1 above)
Comment (dnlkoch) [inline src/plugin/Hover.js]: Whoop! You're so right! Removed in 0a0ccfc and 9b8321e.

File: src/plugin/Hover.js (hunk 3 above)
Comment (dnlkoch) [inline src/plugin/Hover.js]: Adjusted in 63950dd.

**Comment by dnlkoch** [comment]
Thank for the review @marcjansen!

I'll merge...

**Pull request merged**